src/gui/properties/speedplotview.cpp

```diff
@@ -118,55 +107,75 @@ namespace
     }
 }
 
-SpeedPlotView::Averager::Averager(int divider, boost::circular_buffer<PointData> &sink)
-    : m_divider(divider)
-    , m_sink(sink)
-    , m_counter(0)
-    , m_accumulator {}
+SpeedPlotView::Averager::Averager(const milliseconds duration, const milliseconds resolution)
+    : m_resolution {resolution}
+    , m_maxDuration {duration}
+    , m_sink {static_cast<DataCircularBuffer::size_type>(duration / resolution) + 1}
```

Comment:
Why `+ 1`? It doesn't contain values for some time points but for time intervals.

Reply:
for peace of mind

Reply:
>for peace of mind

🙂
You don't still need to play it safe. The maximum size here is quite obvious (after a little analysis). But the extra `+ 1` can be confusing to future reviewers.